flight/core/scheduler/loop.py

```diff
@@ -165,16 +166,22 @@ def __init__(self, debug=False):
         else:
             self._debug = lambda *arg, **kwargs: None
 
+    def enable_debug_logging(self):
+        self.debug = True
+        self._debug = print
+        print("Task scheduler debug logging enabled")
+
     def add_task(self, awaitable_task, priority):
         """
         Add a concurrent task (known as a coroutine, implemented as a generator in CircuitPython)
         Use:
           scheduler.add_task( my_async_method() )
         :param awaitable_task:  The coroutine to be concurrently driven to completion.
         """
-        self._debug("adding task ", awaitable_task)
+        if self.debug:

```

Comment:
It seems unnecessary to me to have 
`if self.debug: self._debug("...")`
as `self._debug` will only print if self.debug is True as it is set to the "do nothing" lambda function on line 167 if self.debug is false.

Reply:
basically `self._debug` will only ever print if `self.debug` is true so there's no point in checking if `self.debug` is True.

Reply:
Did a benchmark on the board and checking a flag is faster than the no-op lambda (surprisingly). This was the reason for the addition of the flag, esp for flight code where the scheduler must be as fast as possible. 

Now, the self.debug attribute fct is useless and can be replaced by print statements only, which will be done in the next commit.